src/main/cluster.ts

```diff
@@ -643,17 +647,26 @@ export class Cluster implements ClusterModel, ClusterState {
       if (!this.allowedNamespaces.length) {
         return [];
       }
-      const resourceAccessStatuses = await Promise.all(
-        apiResources.map(apiResource => this.canI({
-          resource: apiResource.resource,
-          group: apiResource.group,
-          verb: "list",
-          namespace: this.allowedNamespaces[0]
-        }))
-      );
+
+      const resources = apiResources.filter((resource) => this.resourceAccessStatuses.get(resource) === undefined);
+
+      for (const apiResource of resources) {
+        for (const namespace of this.allowedNamespaces.slice(0, 10)) {
+          if (!this.resourceAccessStatuses.get(apiResource)) {
+            const result = await this.canI({
+              resource: apiResource.resource,
+              group: apiResource.group,
+              verb: "list",
+              namespace
+            });
+
+            this.resourceAccessStatuses.set(apiResource, result);
```

Comment:
This is quite broken, since it is "greedy" and is no better then assuming that this first namespace is the "namespace to rule all namespaces". The only benefit I can see from trying multiple namespaces in this fashion is being able to find resources that only exist in some namespaaces.

Reply:
I disagree. Going through multiple namespaces is better than checking just the first one.